httplib.h

```diff
@@ -2859,8 +2859,7 @@ inline bool mmap::open(const char *path) {
   hMapping_ =
       ::CreateFileMappingFromApp(hFile_, NULL, PAGE_READONLY, size_, NULL);
```

Comment:
For consistency, `size_` can be 0 as well.

Reply:
For `CreateFileMappingFromApp`, the documentation doesn't state that size can be 0, so I would continue passing `size_` to stay aligned with the documentation : https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-createfilemappingfromapp

For `CreateFileMappingW`, the documentation explictly states "If this parameter and dwMaximumSizeHigh are 0 (zero), the maximum size of the file mapping object is equal to the current size of the file that hFile identifies." https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-createfilemappingw

Reply:
I got it, thanks!

Reply:
How do you come to this code https://github.com/yhirose/cpp-httplib/pull/1902/files#diff-2324f73f4a4958b756eb44d0607268befc5de62db01164ccd73e154da808d8f3R2849-R2861? I think this blocked is never reached by a compiler if Windows SDK is updated (latest one w/ Windows 7).

I would remove this block completely and deal with bit shifts at `LARGE_INTEGER::HighPart`. `LARGE_INTEGER` is supported on Windows XP, `LARGE_INTEGER::QuadPart` is not supported.

Reply:
According to my SDK's `fileapi.h`:

`GetFileSize` is available if `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)`

`GetFileSizeEx` is available if `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES)`

So the else block would be used if only `WINAPI_PARTITION_DESKTOP` would be set.